src/ssl.c

```diff
@@ -7651,6 +7651,10 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
     #if defined(HAVE_ECC) || defined(HAVE_ED25519) || defined(HAVE_ED448) || \
         (defined(HAVE_PQC) && defined(HAVE_LIBOQS)) || !defined(NO_RSA)
         if (ssl) {
+            ssl->options.haveRSA = 0;
+            ssl->options.haveECC = 0;
+            ssl->options.haveFalconSig = 0;
+            ssl->options.haveDilithiumSig = 0;
```

Comment:
Does this also need to happen in the else statement here with ctx? This needs to get a regression test to maintain it going forward.

Reply:
ctx should stay the same or bad things will happen. I'll add in a test case.